app/src/main/java/com/spascoding/taskycourse/feature_register_screen/data/remote/AuthenticationApi.kt

```diff
@@ -0,0 +1,37 @@
+package com.spascoding.taskycourse.feature_register_screen.data.remote
+
+import com.spascoding.taskycourse.feature_register_screen.data.remote.model.LoginRequest
+import com.spascoding.taskycourse.feature_register_screen.data.remote.model.LoginResponse
+import com.spascoding.taskycourse.feature_register_screen.data.remote.model.RefreshTokenRequest
+import com.spascoding.taskycourse.feature_register_screen.data.remote.model.RefreshTokenResponse
+import com.spascoding.taskycourse.feature_register_screen.data.remote.model.RegisterRequest
+import retrofit2.Call
+import retrofit2.http.Body
+import retrofit2.http.GET
+import retrofit2.http.Header
+import retrofit2.http.POST
+
+interface AuthenticationApi {
+    @POST("/register")
+    fun register(
+        @Body body: RegisterRequest
+    ): Call<Void>
+
+    @POST("/login")
+    fun login(
+        @Body body: LoginRequest
+    ): Call<LoginResponse>
```

Comment:
Since your primary way of async programming seems to be coroutines/Flows in this project, I'd recommend to also stick to it with Retrofit. You can just make these functions suspending functions therefore and stop relying on callbacks you'd get if they return `Call` objects. So replacement could look like this:
```
@POST("/login")
suspend fun login(@Body body: LoginRequest): LoginResponse // or no return type for register respectively
```
Then you can catch errors with try/catch:
```
try {
    api.login(...)
} catch(e: HttpException) {
    println(e.status())
} catch(e: IOException) {
    // Likely no internet or similar IO error
}
```